Objects/genericaliasobject.c

```diff
@@ -718,16 +722,26 @@ ga_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     if (!_PyArg_NoKeywords("GenericAlias", kwds)) {
         return NULL;
     }
-    if (!_PyArg_CheckPositional("GenericAlias", PyTuple_GET_SIZE(args), 2, 2)) {
+    if (!_PyArg_CheckPositional("GenericAlias", PyTuple_GET_SIZE(args), 2, 3)) {
         return NULL;
     }
+
     PyObject *origin = PyTuple_GET_ITEM(args, 0);
     PyObject *arguments = PyTuple_GET_ITEM(args, 1);
+
+    bool starred;
+    if (PyTuple_Size(args) < 3) {
+        starred = false;
+    } else {
+        PyObject *py_starred = PyTuple_GET_ITEM(args, 2);
+        starred = PyLong_AsLong(py_starred);
```

Comment:
Need to check the return value here, and also add a test (what happens if I do `GenericAlias(tuple, int, some_random_object)`?

Reply:
Ah, good point. Done.